src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1236,7 +1236,21 @@ private static string GetUnmappedMemberHandlingAsStr(JsonUnmappedMemberHandling
             private static string GetCreateValueInfoMethodRef(string typeCompilableName) => $"{CreateValueInfoMethodName}<{typeCompilableName}>";
 
             private static string FormatBool(bool value) => value ? "true" : "false";
-            private static string FormatStringLiteral(string? value) => value is null ? "null" : $"\"{value}\"";
+            private static string FormatStringLiteral(string? value)
+            {
+                string returnValue;
+
+                if (value is null)
+                {
+                    returnValue = "null";
+                }
+                else
+                {
+                    returnValue = SyntaxFactory.Literal(value).ToFullString();
+                }
+
+                return returnValue;
+            }
```

Comment:
Seems like this can be simplified
```suggestion
                => value is null ? "null" : SyntaxFactory.Literal(value).ToFullString();
```

Reply:
Fixed by https://github.com/dotnet/runtime/pull/84688/commits/51fcbb09a8b9c5eee71d151817455c97bf484545